File: parcels/rollo/src/tools/element.js
import { Parse } from "../component/tools/factory";

export function updateElement(...args) {



  return this

}

/* Returns instance factory for HTML element. */
export const element = new Proxy(
  {},
  {
    get(_, tag) {
      return (...args) => {
        const result = document.createElement(tag)
        updateElement.call(result, ...args)
        return result

      }
      
    },
  }
);

File: _history/server/client_code/20260306t1600/element/_element.py
from anvil.js.window import document
from ..tools import on
from ._update_element import update_element


class element:
    """Utility for creating HTML elements.
    NOTE Intended for light use."""

    def __getattr__(self, arg: str) -> callable:
        return self[arg]

    def __getitem__(self, arg: str) -> callable:
        tag, *classes = arg.split(".")

        def create(*args, **kwargs):
            instance = document.createElement(tag)
            if classes:
                instance.classList.add(*classes)
            update_element(instance, *args, **kwargs)

            return instance

        return create

    @property
    def on(self):
        return on


element = element()
